### DISCUSSION TITLE
Filesystem API?

### DISCUSSION THREAD

**Discussion by ZXMushroom63** [copy/v86#858] [Ideas]
Would it be possible to use the local file system so as to seperate storage and memory?

https://developer.chrome.com/articles/file-system-access/

For example, using window.showDirectoryPicker() to choose the directory in which to save the machine, or some other thing along those lines.

**Comment by copy** [comment]
Yes, I think that would be useful. See https://github.com/copy/v86/issues/396. PRs welcome.

**Comment by ProgrammerIn-wonderland** [comment]
I kind of have this working in my own site though it's sort of abstracted from v86 itself 

(Demo at https://anura.pro/, select arch, alpine, or Debian, then open the "files" app and click LocalFS to mount a local folder to the "anura filesystem", it will then be visible in v86 under /root/yourMountFolder/) 

It basically just abuses v86's plan 9, if you want to try it out yourself it uses v86 adapted for filer https://github.com/MercuryWorkshop/v86 and then we have a hacky bridge for filer to localfs https://github.com/MercuryWorkshop/anuraOS/blob/main/src/api/LocalFS.ts which just recreates the filer API

So it's totally possible with current v86 if - you don't mind abusing plan9!

**Comment by ProgrammerIn-wonderland** [comment]
![image](https://github.com/copy/v86/assets/30693865/3204c475-e2c9-4c94-b807-c8692dabcd29)
heres a screenshot of it

**Comment by ProgrammerIn-wonderland** [comment]
terminal here is v86 debian

**Comment by ProgrammerIn-wonderland** [comment]
https://github.com/copy/v86/assets/30693865/1d7261e1-a614-4b7c-aa67-a5a1ef30a99a

small video tutorial since its kinda hard to use (you have to wait for linux to finish booting first and it takes a bit since its all in the browser)
